### ISSUE TITLE
[DataGrid] Toolbar inaccessible from screen reader

### ISSUE THREAD

**Issue by CalebJamesStevens** [mui/mui-x#9403]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

https://github.com/mui/mui-x/assets/3165635/f50f41f5-6824-477e-a3ef-195e90cf4309

https://mui.com/x/react-data-grid/components/#toolbar

This link is showing the issue from your own page, repo should not be needed

### Current behavior 😯

When using a screen reader for navigation I can not access the toolbar and have no indication that it even exists as show in the video, the same issue seem to be present with pagination

### Expected behavior 🤔

I should be able to access the toolbar of the datagrid table somehow, 

### Context 🔦

I'm trying to perform a VPAT for the company i work for and this is causing us to be non-compliant with WCAG standards. This also affects any users that use screen readers to navigate a page.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
System:
    OS: macOS 13.0.1
  Binaries:
    Node: 18.16.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.5.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 113.0.5672.92
    Edge: 113.0.1774.57
    Firefox: Not Found
    Safari: 16.1
  npmPackages:
    @emotion/react: ^11.10.4 => 11.11.0 
    @emotion/styled: ^11.10.4 => 11.11.0 
    @mui/base:  5.0.0-alpha.128 
    @mui/core-downloads-tracker:  5.12.3 
    @mui/icons-material: ^5.8.4 => 5.11.16 
    @mui/lab: ^5.0.0-alpha.95 => 5.0.0-alpha.129 
    @mui/material: ^5.10.1 => 5.12.3 
    @mui/private-theming:  5.12.3 
    @mui/styled-engine:  5.12.3 
    @mui/styles: ^5.9.3 => 5.12.3 
    @mui/system:  5.12.3 
    @mui/types:  7.2.4 
    @mui/utils:  5.12.3 
    @mui/x-data-grid:  6.3.1 
    @mui/x-data-grid-pro: ^6.0.4 => 6.3.1 
    @mui/x-date-pickers:  5.0.20 
    @mui/x-date-pickers-pro: ^5.0.0-beta.7 => 5.0.20 
    @mui/x-license-pro:  5.17.12 
    @types/react: ^18.0.28 => 18.2.6 
    react: ^18.2.0 => 18.2.0 
    react-dom: ^18.2.0 => 18.2.0 
    typescript: ^4.9.5 => 4.9.5 
```
</details>

### Order ID or Support key 💳 (optional)
56576

**Comment by romgrk** [comment]
Can't reproduce with the Gnome screen reader on Linux.

@mui/xgrid Can someone on macOS confirm this issue?

**Comment by romgrk** [comment]
@CalebJamesStevens if you're using SR structural navigation, can you confirm which keys/commands you're using to navigate the document?

**Comment by CalebJamesStevens** [comment]
@romgrk any movement on this?

**Comment by cherniavskii** [comment]
I can reproduce it on MacOS.
I believe the issue here is the same as https://github.com/mui/mui-x/issues/8525

**Comment by CalebJamesStevens** [comment]
@cherniavskii could we confirm if this is also happening with pagination?

**Comment by cherniavskii** [comment]
Yes, pagination is also affected.

**Comment by oliviertassinari** [comment]
@CalebJamesStevens can you confirm that the issue is solved in https://github.com/mui/mui-x/pull/9496 You can findba preview link in thid PR description.

**Comment by CalebJamesStevens** [comment]
> @CalebJamesStevens can you confirm that the issue is solved in #9496 You can findba preview link in thid PR description.

I personally am not seeing the issue resolved by this PR link:
https://deploy-preview-9496--material-ui-x.netlify.app/x/react-data-grid/components/#toolbar

Are you?

**Comment by CalebJamesStevens** [comment]
> @CalebJamesStevens can you confirm that the issue is solved in #9496 You can findba preview link in thid PR description.

I apologize i was looking in the wrong place, yes the experimental flag does seem to resolve this issue
